modules/transaction-payment/src/lib.rs

```diff
@@ -308,13 +334,43 @@ pub mod module {
 	pub type AlternativeFeeSwapPath<T: Config> =
 		StorageMap<_, Twox64Concat, T::AccountId, BoundedVec<CurrencyId, T::TradingPathLimit>, OptionQuery>;
 
+	/// The ratio of foreign_asset_per_second to native_asset_per_second.
+	/// the reverse ratio means the value of foreign asset to native asset.
+	#[pallet::storage]
+	#[pallet::getter(fn charge_fee_pool)]
+	pub type FeeRateOfToken<T: Config> = StorageMap<_, Twox64Concat, CurrencyId, Ratio, OptionQuery>;
+
+	/// The treasury fee pool sub account of foregign asset.
+	#[pallet::storage]
+	#[pallet::getter(fn treasury_accounts)]
+	pub type TreasuryAccounts<T: Config> = StorageMap<_, Twox64Concat, CurrencyId, T::AccountId, ValueQuery>;
```

Comment:
no need to store those. can just create sub account on the fly

https://github.com/paritytech/substrate/blob/a64f702d20fd0e979c69d11b60e0d6d6b81d6c6d/frame/bounties/src/lib.rs#L771-L775